client/src/components/ViewPostDialog.tsx

```diff
@@ -0,0 +1,222 @@
+import React from "react";
+import Button from "@mui/material/Button";
+import Dialog from "@mui/material/Dialog";
+import AppBar from "@mui/material/AppBar";
+import IconButton from "@mui/material/IconButton";
+import Typography from "@mui/material/Typography";
+import ArrowBack from "@mui/icons-material/ArrowBack";
+import Slide from "@mui/material/Slide";
+import { TransitionProps } from "@mui/material/transitions";
+import MoreVert from "@mui/icons-material/MoreVert";
+import Menu from "@mui/material/Menu";
+import MenuItem from "@mui/material/MenuItem/MenuItem";
+import Stack from "@mui/material/Stack/Stack";
+import Box from "@mui/material/Box";
+import { styled } from "@mui/material/styles";
+import Paper from "@mui/material/Paper/Paper";
+import LinearProgress from "@mui/material/LinearProgress";
+import TextField from "@mui/material/TextField";
+
+const Transition = React.forwardRef(function Transition(
+  props: TransitionProps & {
+    children: React.ReactElement;
+  },
+  ref: React.Ref<unknown>
+) {
+  return <Slide direction="up" ref={ref} {...props} />;
+});
```

Comment:
What's the purpose of the ref here and can we get away with not using it?

Reply:
I'm not actually sure with this one, the documentation for Dialog transitions use refs. I got rid of them in my code and the transition still seems to work, i'll test it to make sure nothing broke.

Reply:
I had to add the ref back, it was making tests fail. I was following the MUI docs for making transitions and they used this type of transition function, I am not sure how to make it work without the ref